pkg/engine/availabilitysets.go

```diff
@@ -40,6 +36,11 @@ func CreateAvailabilitySet(cs *api.ContainerService, isManagedDisks bool) Availa
 		}
 	}
 
+	if cs.Properties.MasterProfile.PlatformFaultDomainCount != nil {
+		p := int32(*cs.Properties.MasterProfile.PlatformFaultDomainCount)
```

Comment:
For agents the equivalent value assignment is be inside the `profile.IsManagedDisks() {` if statement, what's the different between master and agent?

Reply:
managed disks integrate with the VM faultdomains. they align on the same FD.
UnManaged disks can support less FD and don't guarantee alignment.
So you should not use FD with unmanaged disks really.
But I don't think that silently swallowing the setting is the right approach. if you set FD and you use UnManaged Disks, you should know what you're doing.
Finally, who still uses unmanaged disks? :)

Reply:
I didn't move the agent one because I didn't want to change its behavior, but I would say that it would be better to move it out of the if else.
It's pretty much 0 impact anyway, because who still uses unmanaged disks?